docs/DeveloperGuide.md

```diff
@@ -1,34 +1,426 @@
+ <br>
+ * `Add the scheduled tasks to current list of tasks`
+ 
+ ![Sub Diagram 3](images/Schedule_Sub3.png)
+ <div>Figure 8. Sub Diagram 3</div>
+ <br> 
+ 
+#### 3.1.2 Alternatives
+Aspect : How to capture user's requirements and handle it.
+
+ - Alternative 1 (Current Choice) : Create a separate SchedulableTask object to store user's requirements for each task 
+                                    to be scheduled.
+      - Pros : New class object can be customised without affecting core functionality of the product if the attributes provided by 
+      the user (to tasks to be scheduled) changes.
+      - Cons : New methods similar to previously implemented methods have to be created instead of being reused.
+      
+ - Alternative 2 : Modify existing event or deadline class to model user's requirements.
+      - Pros : It reuses methods and reduces coupling between classes in the TaskComponent.
+      - Cons : Major changes would be required throughout various components of the application to accommodate the new feature. 
+
+     Finally the first alternative was implemented as it was easier to implement and maintain it if changes were required to 
+the user's requirements (e.g. Changes to the number of parameters provided by the user). 
+
+[comment]: # (@@author NizarMohd)
+
+### 3.2. Listing Study Areas 
+
+#### 3.2.1 Implementation 
+The Study Area search is facilitated by StudyAreaList. In this class, it has the list of all existing Study Areas, 
+stored internally as an ArrayList. This ArrayList is called studyAreaList. 
+
+To list the Study Area, we iterate through the entire list of all existing Study Area and conduct a check. The check is 
+as follows :
+
+If in each Study Area, the Study Area meets all the flags stated by the User, the Study Area is then added to the 
+list of available Study Area. If not, the iteration is skipped and move on to the subsequent Study Area.
+
+To do as mentioned above, the StudyAreaList class implements the following main methods :
+
+- searchList() : 
+    - This method will iterate through all the Study Areas in the ArrayList studyAreaList while calling other methods 
+    to carry out the search. Once process has ended, the method will return a list of the Study Area, availStudyArea, 
+    that meets the User requirement.
+- getFlagsInfo() : 
+    - This method will retrieve the flags entered by User, at the same time validating the User input. The criteria 
+    entered by the User will be returned as flags, an Array of Strings of fixed size 5 as the software only supports 5 
+    flags in version 2.
+- isAvailStudyArea() : 
+    - This method does the check mentioned above and returns a boolean value. Based on the flags detected, 
+    the method will iterate through the flags array and check if the Study Area's attribute matches each flag stated in
+    flags. If isAvailStudyArea returns as true for all flags, the Study Area is then added to the output list, 
+    availStudyArea. 
+
```

Comment:
Maybe some of these descriptions should be placed between the diagrams for diagram explanation.